src/api_secret.rs

```diff
@@ -0,0 +1,102 @@
+use std::{
+    collections::BTreeSet,
+    fmt::{self, Display},
+    str::FromStr,
+};
+
+use iroh::{EndpointAddr, EndpointId, SecretKey, TransportAddr};
+use iroh_tickets::{ParseError, Ticket};
+use serde::{Deserialize, Serialize};
+
+/// The secret material used to connect your n0des.iroh.computer project. The
+/// value of these should be treated like any other API key: guard them carefully.
+#[derive(Debug, Clone)]
+pub struct ApiSecret {
+    /// ED25519 secret used to construct rcans from
+    pub secret: SecretKey,
+    /// the n0des endpoint to direct requests to
+    pub remote: EndpointAddr,
+}
+
+impl Display for ApiSecret {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", Ticket::serialize(self))
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+struct Variant0ApiSecret {
```

Comment:
rename to `Variant0EndpointAddr`, and inline `addrs: BTreeSet<TransportAddr>` and remove `Variant0AddrInfo`?